// cf-show-resource/dist/custom-elements/index.js
import { attachShadow, h, proxyCustomElement } from '@stencil/core/internal/client';
export { setAssetPath } from '@stencil/core/internal/client';

const cfShowResourceCss = ":host{display:block}";

const MyComponent = class extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        attachShadow(this);
    }
    componentWillLoad() {
        //let json = JSON.parse(this.resource)
    }
    render() {
        return [
            h("div", null, "Show the details of a resource"),
            h("pre", null, this.resource)
        ];
    }
    static get style() { return cfShowResourceCss; }
};

const CfShowResource = /*@__PURE__*/proxyCustomElement(MyComponent, [1,"cf-show-resource",{"resource":[1]}]);
const defineCustomElements = (opts) => {
  if (typeof customElements !== 'undefined') {
    [
      CfShowResource
    ].forEach(cmp => {
      if (!customElements.get(cmp.is)) {
        customElements.define(cmp.is, cmp, opts);
      }
    });
  }
};

export { CfShowResource, defineCustomElements };
